### PR TITLE
Instruction list

### PR DESCRIPTION
**Pull request by pdunshee-phunware**

### CODE REVIEW

File: Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift [hunk 1]
```diff
@@ -110,11 +115,12 @@ extension PWRouteInstruction {
     }
     
     func distanceString() -> String {
-        let roundedDistance = Int(round(distance))
+        let feetDistance = distance * 3.28084 // meters to feet
```
Comment (peterwangxw) [inline Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift]: Can we add toggle then developer is able to choose meters or feet easily?

File: Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift [hunk 2]
```diff
@@ -14,18 +14,23 @@ extension PWRouteInstruction {
         return .systemFont(ofSize: 15.0, weight: .bold)
     }
     
+    func instructionStringForUser() -> String {
+        return attributedInstructionString(highlightedTextColor: .black, regularTextColor: .black).string
+    }
+    
     func attributedInstructionString(highlightedTextColor: UIColor, regularTextColor: UIColor) -> NSAttributedString {
```
Comment (jfullen-phunware) [inline Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift:30]: Should this be private or is it used somewhere else? If it is only used by the above function it should probably be hidden.

File: Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift (hunk 1 above)
Comment (jfullen-phunware) [inline Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift]: Should this be defined as a constant?

File: Samples/MapScenarios/MapScenarios/Scenarios/Route Instruction List/RouteInstructionListCell.swift [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import PWMapKit
+
+class RouteInstructionListCell: UITableViewCell {
+    
+    @IBOutlet weak var instructionImageView: UIImageView!
+    @IBOutlet weak var instructionLabel: UILabel!
+    
+    func configureWithRouteInstruction(_ routeInstruction: PWRouteInstruction) {
+        instructionImageView.image = routeInstruction.imageFromDirection()
+        instructionLabel.attributedText = routeInstruction.attributedInstructionString(highlightedTextColor: .nasa, regularTextColor: .darkText)
```
Comment (jfullen-phunware) [inline Samples/MapScenarios/MapScenarios/Scenarios/Route Instruction List/RouteInstructionListCell.swift:18]: Love how simple this is!

**Review by jfullen-phunware — COMMENTED** [review]
Left some comments in the code.

File: Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift (hunk 2 above)
Comment (pdunshee-phunware) [reply Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift:30]: Used in RouteInstructionCell and TurnByTurnInstructionCollectionViewCell as well

**Review by peterwangxw — APPROVED** [review]

File: Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift (hunk 1 above)
Comment (jfullen-phunware) [reply Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift]: Or ideally use the device's locale to default to feet or meters. We can also allow them to choose but I feel like the device's locale default would be sufficient for most cases.

File: Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift (hunk 2 above)
Comment (jfullen-phunware) [reply Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift:30]: I was talking about line 21. I know the line 17 function would be called by external objects but I didn't see any references to "attributedInstructionString" outside of this extension.

Comment (pdunshee-phunware) [reply Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift:30]: yeah i was talking about line 21 as well, it's used by those cells
image

Comment (jfullen-phunware) [reply Samples/MapScenarios/MapScenarios/Extensions/PWRouteInstruction+Helper.swift:30]: Ok, got it. Stand corrected.

**Review by jfullen-phunware — APPROVED** [review]

**Review by jfullen-phunware — APPROVED** [review]

**Pull request merged**